executor/aggfuncs/func_json_objectagg.go

```diff
@@ -97,12 +104,45 @@ func (e *jsonObjectAgg) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup
 		realVal := value.Clone().GetValue()
 		switch x := realVal.(type) {
 		case nil, bool, int64, uint64, float64, string, json.BinaryJSON, *types.MyDecimal, []uint8, types.Time, types.Duration:
+			if _, ok := p.entries[keyString]; !ok {
+				memDelta += int64(len(keyString))
+				memDelta += getValMemDelta(realVal)
+			}
 			p.entries[keyString] = realVal
 		default:
 			return 0, json.ErrUnsupportedSecondArgumentType.GenWithStackByArgs(x)
 		}
 	}
-	return 0, nil
+	return memDelta, nil
+}
+
+func getValMemDelta(val interface{}) (memDelta int64) {
+	memDelta = DefInterfaceSize
+	switch v := val.(type) {
+	case nil:
+		return memDelta
+	case bool:
+		memDelta += DefBoolSize
+	case int64:
+		memDelta += DefInt64Size
+	case uint64:
+		memDelta += DefUint64Size
+	case float64:
+		memDelta += DefFloat64Size
+	case string:
+		memDelta += int64(len(v))
+	case json.BinaryJSON:
+		memDelta += int64(unsafe.Sizeof(v))
```

Comment:
Should we using `memDelta += int64(len(v.Value))` here?

Reply:
@b41sh think we should not remove the `srcChk.AppendDatum(0, &types.Datum{})` in aggfun_test.go.

Reply:
I think v.TypeCode also should be considered here.

Reply:
Yes, the `TypeCode` need to be considered also.

Reply:
May we should refer to the code in `distinctUpdateMemDeltaGens`: `jsonVal := row.GetJSON(0)
			bytes := make([]byte, 0)
			bytes = append(bytes, jsonVal.TypeCode)
			bytes = append(bytes, jsonVal.Value...)
			val = string(bytes)
			memDelta = int64(len(val))`, if we do this, maybe we should modify the code in `func_json_objectagg` too.

Reply:
yes, note that we don't need to convert bytes to string
```
        jsonVal := row.GetJSON(0)
	bytes := make([]byte, 0)
	bytes = append(bytes, jsonVal.TypeCode)
	bytes = append(bytes, jsonVal.Value...)
	memDelta = int64(len(bytes))
```